component/azstorage/block_blob.go

```diff
@@ -606,10 +607,35 @@ func (bb *BlockBlob) WriteFromFile(name string, metadata map[string]string, fi *
 	//defer exectime.StatTimeCurrentBlock("WriteFromFile::WriteFromFile")()
 
 	blobURL := bb.Container.NewBlockBlobURL(filepath.Join(bb.Config.prefixPath, name))
-
 	defer log.TimeTrack(time.Now(), "BlockBlob::WriteFromFile", name)
+	blockSize := bb.Config.blockSize
+	// if the block size is not set then we configure it based on file size
+	if !config.IsSet(compName + ".block-size-mb") {
+		// get the size of the file
+		stat, err := fi.Stat()
```

Comment:
Do we need to do this fi.Stat() before upload. Just check if we can pass down the file size in upload request itself from file-cache or something.

Reply:
We shouldnt assume anyone can give us the file size except for libfuse since a customer can one day just use libfuse azstorage right?